util/httputil/http_util.go

```diff
@@ -128,6 +128,8 @@ func GuessKialiURL(r *http.Request) string {
 	// take priority.
 	if fwdPort, ok := r.Header["X-Forwarded-Port"]; ok && len(fwdPort) == 1 {
 		port = fwdPort[0]
+	} else if len(cfg.Server.WebPort) > 0 {
+		port = cfg.Server.WebPort
```

Comment:
Can you, please make this the setting with the most priority?
I see that if both have the `X-Forwarded-Port` header is present and also the manual config is present, the value at the HTTP headers are taken. This would be misleading. I think whatever is present in the config should take priority over any other setting.

Reply:
In general I think that runtime configuration should have higher priority than the one on the configuration file, but from what I've seen elsewhere we're not doing this for other keys, so changed. I've also added a test to cover this case.

Reply:
I agree that in general runtime would override the static config.  But in this case it seems the static config would only be set in order to force the web port to that value.  Sorry to stick my head in here, hopefully I'm not off base.

Reply:
It is important to note that the issue that triggered this change happened because the webserver did not report the forwarded for headers. In this case, even if the configuration conflicted, the proxy should have the higher priority.

However, I do think that consistency is more important than being intuitive.

Reply:
My perception was that people were preferring to have a configuration option rather than configure the HTTP headers in the proxy (or that's what I understand from this comment: https://github.com/kiali/kiali/issues/3111#issuecomment-673056301 -- I also remember one or two other people asking for this option.). It's the main reason why I think the "static" config should take priority.

But well... we can argue :)

Reply:
We all agree here, only discussing semantics :)